lib/vm/src/trap/traphandlers.rs

```diff
@@ -587,8 +686,14 @@ thread_local! {
 /// from traps.
 struct TrapHandlerContext {
     inner: *const u8,
-    handle_trap:
-        fn(*const u8, usize, usize, Option<usize>, &mut dyn FnMut(TrapHandlerRegs)) -> bool,
+    handle_trap: fn(
+        *const u8,
+        usize,
+        usize,
+        Option<usize>,
+        Option<TrapCode>,
```

Comment:
To avoid code duplication, could you pass a `maybe_invalid_inst_address: Option<usize>` to the common trap handling code and do the instruction decoding there?

Reply:
I don't like the idea of have an unprocess address going through all those call. I can simply create a local function to process the illegal intruction and extract the trap information to avoid code duplication, and keep the current mecanism with the `Option<TrapCode>` that for me makes more sense.
What do you think @Amanieu ?

Reply:
Sure, that is fine too.